### PR TITLE
Removed Sensitive Mode

### PR DESCRIPTION
**Pull request by ThePotatoGuy**
**NOTE: I will be removing the word guesser assets in a future release to avoid crashes**

# Key Changes
* Removes sensitive mode setting 
    * the `persistent` value is forced to False now
    * the event property is dummied out (but will still exist in current data)
        * this includes all of the Event filter functions that include sensitive filtering - the functions will still accept sensitive params but they will not be respected
* Added new APIs for name-related easter eggs
    * `persistent._mas_disable_eggs` - primary flag for name easter egg enabled/disabled
    * `mas_egg_manager.is_eggable_name` - checks if a given (lowercase) string is an eggable one (aka one of the girls' names)
    * `mas_egg_manager.name_eggs_enabled` - checks if name easter eggs are enabled/disabled - **use this instead of the primary flag**
    * `mas_egg_manager.natsuki_enabled` - checks if the natsuki name-based easter eggs should be enabled
    * `mas_egg_manager.sayori_enabled` - checks if the sayori name-based easter eggs should be enabled
    * `mas_egg_manager.yuri_enabled` - checks if the yuri name-based easter eggs should be enabled
* refactored the name change label - uses suggestion from #7128 
    * uses the new APIs and also added the confirmation menu asking the player if an easter egg is their real name
* New API for the `cares_about_dokis` flag - `mas_safeToRefDokis` - Checks if the user is okay with potentially sensitive doki references by actually checking if the user has responded to the question.

### Various sensitive changes
* `mas_canShowRisque` - removed sensitive check
* pong uses the new API for the natsuki scare
* Sensitive Mode checkbox removed from settings screen
* ch30_autoload uses the new API for yuri scare
* sayori references when unlocking hangman are now guarded by the new safeToRef API
* removed sensitive check for the sayori lightning in ch30 loop
* replaced sensitive check guards for name scare labels with new API 
* removed sensitive mode check in opendoor greeting ("...Gently open the door" vs "open the door")
* hungry mood uses the new API for natsuki scare
* all "word guesser" hangman replacements removed
* removed sensitive checks in short story menus
* music menu uses the new API for sayori scare

### topics modified
* `monika_sayori`
    * "hung up over it" checks
        * removed sensitive check
        * replaced cares about dokis with `mas_safeToRefDokis` check
    * "sayori's last moments" - removed sensitive check
* `monika_lastpoem` - replaced sensitive check with `mas_safeToRefDokis` - ("Yuri killed herself")
* `monika_tsundere` - removed sensitive check - (references natsuki's home situation, but not really in a negative way)
* `monka_othergames` - ("Yuri's death mattered")
    * removed sensitive check
    * replaced cares about dokis check with `mas_safeToRefDokis` check
* `monika_yellowwp`
    * replaced sensitive mode + cares about dokis check with `mas_safeToRefDokis` check - ("Sayori liked that story too")
    * replaced sensitive mode check with `mas_safeToRefDokis` check - ("poster in the club room")
* `monika_bullying` - ("Sayori's suicide")
    * replaced sensitive mode + cares about dokis check wtih `mas_safeToRefDokis` check
* `monika_load_custom_music` - replaced sayori name check with new API

# Testing

### name eggs (for reference)
* Sayori
    * the music turns on (if off) and is forced to sayonara
    * music menu does not allow changing music
    * volume decrease hotkeys dont work
    * sayori appears in the window for hangman
    * losing at hangman causes the glitch sequence
* Natsuki
    * glitch sequence in the "I'm hungry" mood
    * playing pong while "Okay Everyone" is playing will change the music to the Natsuki version
    * finishing a game of pong launches the neck crack sequence
* Yuri
    * glitch sequence when launching the game (assuming non holidays, and returning Monika to spaceroom/missing Monika)

### tests
* Verify that changing your name to one of the 3 dokis, but responding with "Yes that is my name" when Monika asks, disables all the easter eggs for that doki
     * Verify that responding with "Maybe..." will continue to trigger the easter eggs as normal (even after previously saying it was your name).
* Verify Sensitive Mode is gone from the settings screen
* Verify hangman does not use the word guesser assets
* Verify that all topics with sensitive mode changes work as expected (see above for the specific topics)
* Verify all other places that were touched work as expected now (see above)

### CODE REVIEW

File: Monika After Story/game/script-topics.rpy [hunk 1]
```diff
@@ -1005,42 +999,40 @@ label monika_sayori:
     m 2lsc "I still wish I could have handled that whole thing a little more tactfully."
 
     if (
-            not persistent._mas_sensitive_mode
-            and not renpy.seen_label("monika_sayori")
-            and not persistent._mas_pm_cares_about_dokis
+            not renpy.seen_label("monika_sayori")
```
Comment (Legendkiller21) [inline Monika After Story/game/script-topics.rpy:1002]:
```suggestion
            not mas_getEVL_shown_count("monika_sayori")
```
The label is marked as seen the moment we get in so we should check the `shown_count` in this one

File: Monika After Story/game/script-topics.rpy [hunk 2]
```diff
@@ -1656,7 +1648,7 @@ init 5 python:
 
 label monika_lastpoem:
     m 3eud "Hey, you remember that last poem I showed you?"
-    if persistent._mas_sensitive_mode:
+    if mas_safeToRefDokis():
```
Comment (Legendkiller21) [inline Monika After Story/game/script-topics.rpy:1651]:
```suggestion
    if not mas_safeToRefDokis():
```

**Review by Legendkiller21 — CHANGES_REQUESTED** [review]
Some small stuff

**Review by Legendkiller21 — APPROVED** [review]

**Review by multimokia — APPROVED** [review]

**Comment by ThePotatoGuy** [comment]
@jmwall24 can you finish your review on this before the end of the week? I want to unstable this soon.

**Comment by jmwall24** [comment]
Sorry, I reviewed this last week, forgot to actually submit the review.

**Review by jmwall24 — APPROVED** [review]

**Comment by multimokia** [comment]
au revoir, sensitive.

**Pull request merged**